src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,122 @@
+package duke.task;
+
+import org.json.simple.JSONObject;
+
+import duke.date.Date;
+import duke.exception.BadInputFormatException;
+import duke.exception.InvalidDateException;
+
+/** Represents a Deadline object.*/
+public class Deadline extends DatedTask {
+    /**
+     * Deadline constructor.
+     *
+     * @param description The Deadline's description.
+     * @throws BadInputFormatException If the deadline is badly formatted.
+     */
+    Deadline(String description) throws BadInputFormatException, InvalidDateException {
+        this(parseToDescription(description), parseToDate(description));
+    }
+
+    /**
+     * Deadline constructor.
+     *
+     * @param description The Deadline's description.
+     * @param by The date to complete the task by.
+     */
+    Deadline(String description, Date by) {
+        super(description, by);
+    }
+
+    /**
+     * Deadline constructor.
+     *
+     * @param description The Deadline's description.
+     * @param by The date to complete the task by.
+     * @param isDone The done status of the Deadline.
+     */
+    private Deadline(String description, Date by, boolean isDone) {
+        super(description, by, isDone);
+    }
+
+    /**
+     * Factory Deadline method.
+     *
+     * @param description The user's input.
+     * @return A new Deadline object.
+     */
+    public static Deadline of(String description) throws BadInputFormatException, InvalidDateException {
+        return new Deadline(description);
+    }
+
+    /**
+     * Factory Deadline method.
+     *
+     * @param description The Deadline's description.
+     * @param by The date to complete the Task by.
+     * @param isDone The done status of the Task
```

Comment:
I think you may have forgotten to add a punctuation behind this parameter description. Maybe you could add it in.

Reply:
Nice catch.